client/src/toplevels/Toplevel.res

```diff
@@ -137,16 +143,54 @@ let spaceOfHandler = (h: PT.Handler.t): handlerSpace => SpecHeaders.spaceOf(h.sp
 let spaceOf = (tl: toplevel): option<handlerSpace> =>
   tl |> asHandler |> Option.map(~f=spaceOfHandler)
 
-let isHTTPHandler = (tl: toplevel): bool => tl |> spaceOf |> \"="(Some(HSHTTP))
-
-let isCronHandler = (tl: toplevel): bool => tl |> spaceOf |> \"="(Some(HSCron))
-
-let isWorkerHandler = (tl: toplevel): bool => tl |> spaceOf |> \"="(Some(HSWorker))
+let isHTTPHandler = (tl: toplevel): bool =>
+  switch asHandler(tl) {
+  | Some(h) =>
+    switch h.spec {
+    | PT.Handler.Spec.HTTP(_) => true
+    | _ => false
+    }
+  | None => false
+  }
```

Comment:
Minor: feels like this could be reduced to something like:

```
let isHTTPHandler = (tl: toplevel): bool =>
  switch Option.map(asHandler(tl), ~f=h => h.spec) {
  | Some(PT.Handler.Spec.HTTP(h)) => true
  | _ => false
  }
```

Reply:
Even better:

```
  switch asHandler(tl) {
  | Some({spec: PT.Handler.Spec.REPL(_), _}) => true
  | _ => false
  }
```